src/components/pacbio/PacbioLibraryCreate.vue

```diff
@@ -125,26 +135,57 @@ export default {
   },
   data() {
     return {
-      library: { tag: { group_id: '' }, sample: {} },
-      tagOptions: [{ value: '', text: 'No tag' }],
+      library: { tag: { id: '' }, sample: {} },
+      selectedTagSetId: '',
+      selectedTagId: '',
     }
   },
   computed: {
-    ...mapGetters('traction', ['tractionTags']),
+    // Map fetched Pacbio tag sets and tags
+    ...mapState('traction/pacbio/poolCreate', {
+      tagSets: (state) => state.resources.tagSets,
+      tags: (state) => state.resources.tags,
+    }),
+    // Return options for the first dropdown
+    tagSetOptions() {
+      const tagSets = Object.values(this.tagSets)
+      const options = tagSets.map(({ id: value, name: text }) => ({ value, text }))
+      const placeholder = { value: '', text: 'Please select a tag set' }
+      return [placeholder, ...options]
+    },
+    // Return options for the second dropdown
+    tagOptions() {
+      const placeholder = { value: '', text: 'Please select a tag' }
```

Comment:
You could do this formatting in the tag fetch when the tag data is returned form the service. E.g tagset state in would look like tagSets: { 'tag_set_example': {  tags: [ {'id': 1, 'group_id' : '123'} ]}}.

Reply:
I explicitly separated the placeholder to make it visible. tagOptions depends on the selectedTagSet in the first dropdown. It has to be computed again.

Reply:
Options are formatted and written to state. The component retrieves the options and adds the placeholders only, which can be customised easily there.

Reply:
I dont think it particularly matters where the options are computed, i was more talking about how the tag set and tags data looks. I think the changes you made are good though and easy enough to change in the future if we need to